realm/realm-jni/build.gradle

```diff
@@ -248,6 +248,72 @@ task deployCore(group: 'build setup', description: 'Deploy the latest version of
     }
 }
 
+task assembleRobolectric(description: "Build the Robolectric shared library for Darwin") {
+    def staticLibPath = "/usr/local/lib/librealm.a"
+
+    if (System.env.REALM_CORE_LIB) {
+        def realmCoreLibDir = file(System.env.REALM_CORE_LIB)
+        staticLibPath = "$realmCoreLibDir/librealm.a"
+    }
+
+    doFirst {
+        if (file(staticLibPath).exists() == false) {
+            throw new GradleException("The Realm Core library for OS X is not installed correctly. If the static lib is not located in /usr/local/lib, please set the environment variable REALM_CORE_LIB")
+        }
+    }
+
+    doLast {
+        copy {
+            from staticLibPath
+            into "${projectDir}/../realm-library/src"
+        }
+
+        exec {
+            environment CC: "${clang?'clang':'gcc'}"
+            environment STRIP: "/usr/bin/strip -o librealm-jni-darwin-stripped.dylib"
+            commandLine = [
+                'make',
+                "-j${Runtime.getRuntime().availableProcessors() * 2}",
+                "-l${Runtime.getRuntime().availableProcessors()}",
+                '-C', "${projectDir}/src",
+                "CC_IS=${clang?'clang':'gcc'}",
+                "REALM_CFLAGS_COMMON=-Wno-variadic-macros -DREALM_HAVE_CONFIG -DPIC -I/usr/local/include/realm/ -I/usr/local/include/ -I/System/Library/Frameworks/JavaVM.framework/Headers/",
+                "CFLAGS_ARCH=${commonCflags.join(' ')}",
+                "BASE_DENOM=darwin",
+                "REALM_LDFLAGS_COMMON=-lrealm -lstdc++ -L${projectDir}/../realm-library/src -L/usr/local/lib/ -Wl,-dead_strip -flto -Wl,-search_paths_first -framework CoreServices",
+                'LIB_SUFFIX_SHARED=.dylib',
+                "librealm-jni-darwin${getDebugExt()}.dylib"
+            ]
+        }
+
+        copy {
+            from "${projectDir}/src/librealm-jni-darwin${getDebugExt()}${getStrippedExt()}.dylib"
+            into "${projectDir}/../realm-library/src/main/robolectricLibs"
+            rename "librealm-jni-darwin${getDebugExt()}${getStrippedExt()}.dylib", 'librealm-jni-darwin.dylib'
+        }
+
+        // Store the unstripped version
+        copy {
+            from "${projectDir}/src/librealm-jni-darwin${getDebugExt()}.dylib"
+            into "${projectDir}/../build/output/jniLibs-unstripped/robolectric"
+            rename "librealm-jni-darwin${getDebugExt()}.dylib", 'librealm-jni-darwin.dylib'
+        }
+
+        delete fileTree(dir: "${projectDir}/../realm-library/src", include: '**/librealm.a')
+    }
+}
+
+task installRobolectric(description: "Install native library to support Robolectric.") {
```

Comment:
It's for development. Without this, we will copy the internal build to its example manually.